File: POOBanco/Java/Cofre.java
package poobanco;

public class Cofre implements Guardado {
    private float total;
    
    @Override
    public void depositar(Valor vl) {
        if(vl.getVale() >= 0) {
            this.total += vl.getVale();
            
            System.out.printf("Foi depositada no cofre a quantia de R$%.2f.\n", vl.getVale());
        }
        else {
            System.out.println("Não é possível depositar esse valor no cofre!");
        }
    }
    
    @Override
    public void retirar(float qt) {
        if(qt >= 0) {
            if(qt > this.total) {
                System.out.printf("Foi retirada do cofre a quantia de R$%.2f.\n", this.total);
                
                this.total -= this.total;
            }
            else {
                System.out.printf("Foi retirada do cofre a quantia de R$%.2f.\n", qt);
                
                this.total -= qt;
            }
        }
        else {
            System.out.println("Não é possível retirar esse valor do cofre!");
        }
    }
    
    @Override
    public void verificar() {
        System.out.printf("O valor total dentro do cofre é de R$%.2f.\n", this.total);
    }
    
    public Cofre() {
        this.total = 0;
    }
    
    public float getTotal() {
        return this.total;
    }
    
    public void setTotal(float total) {
        this.total = total;
    }
}

File: POOBanco/Python/Cofre.py
from Guardado import Guardado

class Cofre(Guardado):
    __total = 0.0

    def depositar(self, vl):
        if vl.getVale() >= 0:
            self.__total += vl.getVale()

            print("Foi depositada no cofre a quantia de R$%.2f." % vl.getVale())
        else:
            print("Não é possível depositar esse valor no cofre!")

    def retirar(self, qt):
        if qt >= 0:
            if qt > self.__total:
                print("Foi retirada do cofre a quantia de R$%.2f." % self.__total)

                self.__total -= self.__total
            else:
                print("Foi retirada do cofre a quantia de R$%.2f." % self.__total)

                self.__total -= qt
        else:
            print("Não é possível retirar esse valor do cofre!")

    def verificar(self):
        print("O valor total dentro do cofre é de R$%.2f." % self.__total)

    def __init__(self):
        self.__total = 0

    def __del__(self):
        print("Objeto Cofre Destruído com Sucesso!")

    def getTotal(self):
        return self.__total

    def setTotal(self, total):
        self.__total = total